akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -60,8 +59,8 @@ private[io] final class AsyncDnsResolver(
         Try {
           val address = InetAddress.getByName(name) // only checks validity, since known to be IP address
           val record = address match {
-            case _: Inet4Address           ⇒ ARecord(name, Int.MaxValue, address)
-            case ipv6address: Inet6Address ⇒ AAAARecord(name, Int.MaxValue, ipv6address)
+            case _: Inet4Address           ⇒ ARecord(name, FiniteCache.effectivelyForever, address)
+            case ipv6address: Inet6Address ⇒ AAAARecord(name, FiniteCache.effectivelyForever, ipv6address)
```

Comment:
When the `name` in the `Lookup` is a valid IP, the query is short-circuited. This use case requires `FiniteCache.effectivelyForever`.